### PR TITLE
ODC-5602-Added step definitions related to eventing features smoke test cases

### PR DESCRIPTION
**Pull request by makambalaji**
**Problem**: Smoke test cases related to eventing operators
**Solution**: Included below step definitions

- actions-on-event-sources.ts
- create-event-sources.ts
- event-sources-sink-to-uri.ts

**Browser and its version**: 
 Chrome  89

**Execution Process**:

1. Cypress file modification [local to plugin folder]

> Update TAGS section in knative-plugin/cypress.json as `@knative-eventing and @smoke and not @manual and not @to-do`

2. Command to execute

> yarn run test-cypress-knative
Select the 
eventing/actions-on-event-sources.feature
eventing/create-event-sources.feature
eventing/event-sources-sink-to-uri.feature

**Screenshots**
image
image
image

### CODE REVIEW

File: frontend/packages/dev-console/integration-tests/support/constants/add.ts [hunk 1]
```diff
@@ -65,3 +66,10 @@ export enum builderImages {
   Java = 'Java',
   NodeJs = 'Node.js',
 }
+
+export enum eventSourceCards {
+  ApiServerSource = 'Api Server Source',
```
Comment (gajanan-more) [inline frontend/packages/dev-console/integration-tests/support/constants/add.ts:71]:
```suggestion
  ApiServerSource = 'ApiServerSource',
```

File: frontend/packages/dev-console/integration-tests/support/constants/global.ts [hunk 2]
```diff
@@ -16,17 +16,20 @@ export enum devNavigationMenu {
...
-  Administrator = ' Administrator',
+  Administrator = 'Administrator',
 }
 
 export enum operators {
   PipelinesOperator = 'Pipeline',
   ServerlessOperator = 'Serverless',
   VirtualizationOperator = 'Virtualization',
-  KnativeCamelOperator = 'Knative Apache Camel K',
+  RedHatCamelKOperator = 'Red Hat Integration - Camel K',
```
Comment (gajanan-more) [inline frontend/packages/dev-console/integration-tests/support/constants/global.ts:26]:
```suggestion
  RedHatIntegrationCamelK = 'Red Hat Integration - Camel K',
```

File: frontend/packages/dev-console/integration-tests/support/constants/global.ts [hunk 3]
```diff
@@ -16,17 +16,20 @@ export enum devNavigationMenu {
...
+  ApacheCamelKOperator = 'Camel K Operator',
+  knativeApacheCamelOperator = 'Knative Apache Camel Operator',
```
Comment (gajanan-more) [inline frontend/packages/dev-console/integration-tests/support/constants/global.ts:28]:
```suggestion
  KnativeApacheCamelOperator = 'Knative Apache Camel Operator',
```

File: frontend/packages/dev-console/integration-tests/support/pages/operators-page.ts [hunk 4]
```diff
@@ -63,9 +70,30 @@ export const operatorsPage = {
...
       }
-      case 'knative Apache Camel Operator':
-      case operators.KnativeCamelOperator: {
-        cy.get(operatorsPO.operatorHub.knativeCamelOperatorCard).click();
+      case 'Red Hat Integration - AMQ Streams':
+      case operators.ApacheKafka: {
+        cy.get(operatorsPO.operatorHub.apacheKafkaOperatorCard).click();
+        break;
+      }
+      case 'RedHat Camel K Operator':
```
Comment (gajanan-more) [inline frontend/packages/dev-console/integration-tests/support/pages/operators-page.ts:78]:
```suggestion
      case 'Red Hat Camel K Operator':
```

File: frontend/packages/knative-plugin/integration-tests/features/eventing/event-sources-sink-to-uri.feature [hunk 5]
```diff
@@ -4,38 +4,28 @@ Feature: Event Sources can able sink to URI as well as Resource
...
+              And user selects Resource option in Sink section
+              And user selects the resource "nodejs-ex-git" for event source
+              And user enters the event source name "sink-binding"
               And user clicks on the Create button
              Then user will be redirected to Topology page
-              And user will see that event source is sinked with selected resource
+              And user will see the event source "sink-binding" is sinked with selected resource "nodejs-ex-git"
 
 
         @regression @manual
```
Comment (gajanan-more) [inline frontend/packages/knative-plugin/integration-tests/features/eventing/event-sources-sink-to-uri.feature:26]:
```suggestion
        @regression, @manual
```

File: frontend/packages/knative-plugin/integration-tests/features/eventing/eventing-channel-subscription.feature [hunk 6]
```diff
@@ -9,14 +9,15 @@ Feature: Knative Eventing Channel and Subscription
...
+             When user right clicks on the Channel "channel-test" to open the context menu
+              And user selects "Add Subscription" from Context Menu
+              And user enters Name as "channel-subscrip" on Add Subscription modal
+              And user selects Subscriber "nodejs-ex-git" on Add Subscription modal
               And user clicks on Add button
-             Then user will see connection between Channel and Subscriber
+             Then user will see connection between Channel "channel-test" and Subscriber "nodejs-ex-git"
 
 
         @regression @manual
```
Comment (gajanan-more) [inline frontend/packages/knative-plugin/integration-tests/features/eventing/eventing-channel-subscription.feature:23]:
```suggestion
        @regression, @manual
```

File: frontend/packages/knative-plugin/integration-tests/features/knative/actions-on-knative-revision.feature [hunk 7]
```diff
@@ -107,8 +91,6 @@ Feature: Perform actions on knative revision
 
         @regression @manual
```
Comment (gajanan-more) [inline frontend/packages/knative-plugin/integration-tests/features/knative/actions-on-knative-revision.feature:92]:
```suggestion
        @regression, @manual
```

File: frontend/packages/knative-plugin/integration-tests/features/knative/actions-on-knative-revision.feature [hunk 8]
```diff
@@ -117,8 +99,6 @@ Feature: Perform actions on knative revision
 
         @smoke @manual
```
Comment (gajanan-more) [inline frontend/packages/knative-plugin/integration-tests/features/knative/actions-on-knative-revision.feature:100]:
```suggestion
        @smoke, @manual
```

**Review by gajanan-more — CHANGES_REQUESTED** [review]

File: frontend/packages/dev-console/integration-tests/support/constants/add.ts (hunk 1 above)
Comment (makambalaji) [reply frontend/packages/dev-console/integration-tests/support/constants/add.ts:71]: event source card name is displaying same as shown below
image

File: frontend/packages/knative-plugin/integration-tests/features/eventing/event-sources-sink-to-uri.feature (hunk 5 above)
Comment (makambalaji) [reply frontend/packages/knative-plugin/integration-tests/features/eventing/event-sources-sink-to-uri.feature:26]: this is expected, or else yarn run gherkin lint will throw the error

File: frontend/packages/knative-plugin/integration-tests/features/eventing/eventing-channel-subscription.feature (hunk 6 above)
Comment (makambalaji) [reply frontend/packages/knative-plugin/integration-tests/features/eventing/eventing-channel-subscription.feature:23]: this is expected, or else yarn run gherkin lint will throw the error

File: frontend/packages/knative-plugin/integration-tests/features/knative/actions-on-knative-revision.feature (hunk 8 above)
Comment (makambalaji) [reply frontend/packages/knative-plugin/integration-tests/features/knative/actions-on-knative-revision.feature:100]: this is expected, or else yarn run gherkin lint will throw the error

File: frontend/packages/dev-console/integration-tests/support/constants/global.ts (hunk 2 above)
Comment (makambalaji) [reply frontend/packages/dev-console/integration-tests/support/constants/global.ts]: updated

File: frontend/packages/dev-console/integration-tests/support/constants/global.ts (hunk 3 above)
Comment (makambalaji) [reply frontend/packages/dev-console/integration-tests/support/constants/global.ts]: updated

File: frontend/packages/dev-console/integration-tests/support/pages/operators-page.ts (hunk 4 above)
Comment (makambalaji) [reply frontend/packages/dev-console/integration-tests/support/pages/operators-page.ts]: updated

File: frontend/packages/knative-plugin/integration-tests/features/eventing/event-sources-sink-to-uri.feature (hunk 5 above)
Comment (makambalaji) [reply frontend/packages/knative-plugin/integration-tests/features/eventing/event-sources-sink-to-uri.feature:10]: updated

**Review by gajanan-more — APPROVED** [review]
Screenshot from 2021-04-05 17-24-14

Working as expected!!!

/lgtm

**Pull request closed without merging**